src/H5FDsubfiling/H5subfiling_common.h

```diff
@@ -0,0 +1,257 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Copyright by The HDF Group.                                               *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of HDF5.  The full HDF5 copyright notice, including     *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the root of the source code       *
+ * distribution tree, or in https://support.hdfgroup.org/ftp/HDF5/releases.  *
+ * If you do not have access to either file, you may request a copy from     *
+ * help@hdfgroup.org.                                                        *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+/*
+ * Header file for shared code between the HDF5 Subfiling VFD and IOC VFD
```

Comment:
Originally, the subfiling and I/O concentrator VFDs were tightly couple together, with most of their routines being marked extern and directly shared between source files. To make the feature more modular and allow subfiling to use different I/O concentrator backends, the VFDs were separated from each other and common subfiling functionality was moved into this "module" so that one can write their own I/O concentrator backend and just include this header.